### PR TITLE
Update matching.py

### PR DESCRIPTION
**Pull request by Ankur-AVP-Patel**
Added O-acetylation support in response to reviewer comments

### CODE REVIEW

**Comment by bobturneruk** [comment]
It would be good if our tests covered the new code, but I guess this can happen later.

**Comment by bobturneruk** [comment]
@Ankur-AVP-Patel should o-acetylation be selectable here?
image

**Comment by Ankur-AVP-Patel** [comment]
> @Ankur-AVP-Patel should o-acetylation be selectable here?
> image

Yes, will create new baseline later - might new MS data as O acetylation is typically seen in gram positive bacteria. Code should be fine though as its just the same as the Deacetyl modification but with the sign for the mass flipped.

**Comment by bobturneruk** [comment]
Need to be consistent in how we use "O-Acety*" case and "acetylated" vs. "Acetylation" etc.

File: pgfinder/matching.py [hunk 1]
```diff
@@ -402,6 +423,10 @@ def data_analysis(raw_data_df: pd.DataFrame, theo_masses_df: pd.DataFrame, rt_wi
         deac_anhy_df = modification_generator(obs_theo_df, "Deacetyl-Anhydro")
     else:
         deac_anhy_df = pd.DataFrame()
+    if 'O-acetylation' in enabled_mod_list:
+        oacetyl_df = modification_generator(obs_theo_df, "O-acetylated")
```
Comment (bobturneruk) [inline pgfinder/matching.py:427]:
```suggestion
        oacetyl_df = modification_generator(obs_theo_df, "O-Acetylated")
```

**Review by bobturneruk — APPROVED** [review]
Looks good! Please consider specific comments and do a "sanity check" once merged to ensure that O-Acetylated masses are being assigned correctly.

**Pull request merged**